### ISSUE TITLE
[Feat] torch implementation of opencv drawing functions

### ISSUE THREAD

**Issue by theRealSuperMario** [kornia/kornia#607]
## 🚀 Feature

I would like to have an equivalent of the opencv drawing functions implemented in torch. The reason for that is simply that I do not want to detach all tensors and iterate over the batch to apply some visualisation. 

## Motivation

I want to the able to just apply the same opencv functionality to torch tensors as I normally do to 2d images.

## Pitch

Just sit down and implement those functions under `kornia.cv2.drawing`. I could get started with the simple ones, (if desired) and make a PR.

## Alternatives

Haven't found any, yet.

## Additional context

https://docs.opencv.org/2.4/modules/core/doc/drawing_functions.html

**Comment by ducha-aiki** [comment]
Sure, go ahead!

**Comment by theRealSuperMario** [comment]
Well I would rather wait for some approval - maybe this feature is not really wanted after all.

**Comment by ducha-aiki** [comment]
I am one of maintainers and I approve it.
 But if you want @edgarriba himself, here you go https://github.com/kornia/kornia/issues/278#issuecomment-549024212
https://github.com/kornia/kornia/issues/434

**Comment by edgarriba** [comment]
@theRealSuperMario sure go ahead. Just 
a couple of considerations, please first make a good API propal and take into account that everything have to be implemented using pytorch operators since we don't wrap any external library.

**Comment by edgarriba** [comment]
vote for `kornia.drawing` or `kornia.render` naming module instead

**Comment by theRealSuperMario** [comment]
sounds like agreeable terms.

**Comment by edgarriba** [comment]
@theRealSuperMario still any interest on this? I think we can start with easy functionalities like drawing lines or rectangles as utility functions in `kornia.utils.image`

I can implement myself for drawing rectangle. /cc @anguelos

**Comment by mmathew23** [comment]
@edgarriba, I'm certainly interested in helping and have capacity as well. Have you started on rectangle yet? I'm happy to use that as a reference and work on the remaining functions.

**Comment by edgarriba** [comment]
@mmathew23 haven't started yet. Feel free to go ahead. One possible API would be `kornia.utils.rectangle(image: torch.Tensor, boxes: torch.Tensor) -> torch.Tensor` where image `BxCxHxW` and boxes `BxNx4`.

**Comment by mmathew23** [comment]
One doubt I have is the differentiable part. Opencv and PIL both write to memory directly when drawing. Going this route allows the color of the drawing to be differentiable but not the location. My initial thought was to use grid_sample or warping, but not sure if there is a better alternative.

**Comment by theRealSuperMario** [comment]
There is No need for differentieability. IT IS Just for visualization.

DoubleMathew <notifications@github.com> schrieb am Di., 20. Okt. 2020,
09:00:

**Comment by edgarriba** [comment]
we can start first with non-differentiable

**Comment by mmathew23** [comment]
Sounds good. Will start with a basic implementation of line and rectangle. My current prototype edits the image tensors in place, which mimics opencv. Is there a preference to operate on a copy instead?

@edgarriba, to clarify, when you mentioned boxes as BxNx4, I assume N refers to the number of boxes you want to draw for the batch?

**Comment by edgarriba** [comment]
@edgarriba yes, N boxes per batch. As for in place or not, I don't have a strong opinion on that but I believe that since the intention for now is to not use it for backprop we can do it in that way.

**Comment by edgarriba** [comment]
closing and continue here https://github.com/kornia/kornia/issues/936
